npc/custom/events/mushroom_event.txt

```diff
@@ -50,9 +50,12 @@ OnMinute10:	// Start time (every hour)
 
 OnMobKilled:
 	set .Spawn, .Spawn - 1;
-	getitem .Prize, .Amount;
-	if (.Spawn) announce "[ "+strcharinfo(PC_NAME)+" ] has killed a Mushroom. There are now "+.Spawn+" Mushroom(s) left.",bc_map;
-	else {
+	if (playerattached()) {
```

Comment:
according to documentations and code this function is a non-boolean, it always return an integer value and 0 on failure, a ```!= 0``` check should be placed instead.

Reply:
done.